.github/workflows/test.yml

```diff
@@ -3,14 +3,39 @@ on: pull_request
 env: { CLICOLOR_FORCE: 1 }
 
 jobs:
-  test:
+  cargo-fmt:
+    name: cargo-fmt@stable
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
       - name: Run formatting checks
         run: cargo fmt --check
+  cargo-clippy:
+    name: cargo-clippy@${{ matrix.version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [1.63, stable, beta]
```

Comment:
There's really no point in running clippy with anything other than stable. The easiest would be to merge this into the current `cargo-fmt` job.

Reply:
Oh your right, that was my logic for only running formatting checks on `stable`, but clippy checks probably need to be the same as well.

I'll probably still make it a separate job because I'd like them to all be ran in parallel, but I can make it only run on stable.